### PR TITLE
IGNITE-16745 fix the bug of SQLGetStmtAttr

### PR DESCRIPTION
**Pull request by alinaliBQ**
fix the error of SQLGetStmtAttr(SQL_ATTR_ROW_ARRAY_SIZE) returning the wrong value for row array size.
Reason for change: the attribute of SQL_ATTR_ROW_ARRAY_SIZE shouldn't be returned as the default 1, as BI tools can try to set the rowArraySize by calling SQLSetStmtAttr. 

I don't see any Jira tickets associated with this bug. This is my first PR to Apache Ignite, please let me know if there's anything that I'm missing here.

Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [x] There is a single JIRA ticket related to the pull request. 
- [x] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [x] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [x] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [x] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

**Comment by isapego** [comment]
Thanks for the report. Seems like an actual bug. Filed a ticket: IGNITE-16745

**Comment by alinaliBQ** [comment]
Thanks for creating the ticket! I've created an account for the Jira board. Would like to assign it to myself but it looks like I don't have access. 
Also, it seems that there is no existing test for `SQLSetStmtAttr`, do I still need to create a test for this fix?

**Comment by alinaliBQ** [comment]
Nevermind, I found some tests for SQLSetStmtAttr and SQLGetStmtAttr, just no checks yet. I can work on those.

File: modules/platforms/cpp/odbc-test/src/api_robustness_test.cpp [hunk 1]
```diff
@@ -795,6 +795,31 @@ BOOST_AUTO_TEST_CASE(TestSQLSetStmtAttr)
...
+    // check that statement array size is set correctly 
+
+    Connect("DRIVER={Apache Ignite};address=127.0.0.1:11110;schema=cache");
+
+    SQLCHAR buffer[ODBC_BUFFER_SIZE];
+    SQLINTEGER resLen = 0;
+
+    // repeat test for different values
+    SQLULEN valList[5] = {10, 52, 81, 103, 304};
+    for (SQLULEN val : valList) {
```
Comment (isapego) [inline modules/platforms/cpp/odbc-test/src/api_robustness_test.cpp:809]: Sorry, we do not use C++11 yet. Can you change this part to index form of for loop (`for (i = 0; ...)`)?

Comment (alinaliBQ) [reply modules/platforms/cpp/odbc-test/src/api_robustness_test.cpp]: no worries. I made the change

**Review by ivandasch — APPROVED** [review]

**Pull request closed without merging**